src/GraphQL.Relay.Test/Utilities/EdgeRangeTests.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using GraphQL.Relay.Utilities;
+using Xunit;
+
+namespace GraphQL.Relay.Test.Utilities
+{
+    public class EdgeRangeTests
+    {
+        [Fact]
+        public void Ctor_WhenStartOffsetIsNegative_ThrowsArgumentOutOfRangeException()
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() => new EdgeRange(-1, 0));
+        }
+
+        [Fact]
+        public void Ctor_WhenEndOffsetIsNegative_ThrowsArgumentOutOfRangeException()
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() => new EdgeRange(0, -1));
+        }
+
+        [Theory]
+        [InlineData(3)]
+        [InlineData(6)]
+        public void Count_WhenStartEqualsEnd_Returns1(int offset)
+        {
+            var range = new EdgeRange(offset, offset);
+            Assert.Equal(1, range.Count);
+        }
+
+        [Theory]
+        [InlineData(10, 9)]
+        [InlineData(10, 8)]
+        public void Ctor_WhenEndOffsetIsLessThanStartOffset_ClampsEndOffsetToOneLessThanStart(int start, int end)
+        {
+            var range = new EdgeRange(start, end);
+            Assert.Equal(start, range.StartOffset);
+            Assert.Equal(start - 1, range.EndOffset);
+            Assert.Equal(0, range.Count);
+            Assert.True(range.IsEmpty);
```

Comment:
when does this happen? and why is it better to handle it this vs a hard error?

Reply:
Handling this case gracefully allows the code to deal with the scenario where `before` < `after`. Those cases should in my opinion just return an empty result and not throw an exception. EdgeRange considers all cases where end < start to be empty. I've add another test case that reflects this.